src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -86,6 +90,31 @@ public Agency getAgency(final String agencyId, final StatusFilter filter, final
                 .map(AgencyTransformer::transform).orElseThrow(EntityNotFoundException.withId(agencyId));
     }
 
+
+    @Transactional
+    public Agency updateAgency(final String agencyId, final Agency agencyToUpdate) {
+        final var agency = agencyLocationRepository.findById(agencyId).orElseThrow(EntityNotFoundException.withId(agencyId));
+        if (!agency.getId().equalsIgnoreCase(agencyToUpdate.getAgencyId())) {
+            throw new BadRequestException(format("Agency ID %s does not match payload ID %s", agencyId, agencyToUpdate.getAgencyId()));
+        }
+
+        final var agencyLocationType = agencyLocationTypeReferenceCodeRepository.findById(new uk.gov.justice.hmpps.prison.repository.jpa.model.ReferenceCode.Pk(AgencyLocationType.AGY_LOC_TYPE, agencyToUpdate.getAgencyType())).orElseThrow(EntityNotFoundException.withMessage(format("Agency Type [%s] not found", agencyToUpdate.getAgencyType())));
+        return AgencyTransformer.transform(AgencyTransformer.update(agency, agencyToUpdate, agencyLocationType));
+    }
+
+    @Transactional
+    public Agency createAgency(final Agency agencyToCreate) {
+        agencyLocationRepository.findById(agencyToCreate.getAgencyId())
```

Comment:
interesting to think where we will get an "ID" from in the future

Reply:
Indeed!